pkg/scheduler/factory/factory.go

```diff
@@ -919,14 +919,17 @@ func (c *configFactory) CreateFromConfig(policy schedulerapi.Policy) (*scheduler
 
 	extenders := make([]algorithm.SchedulerExtender, 0)
 	if len(policy.ExtenderConfigs) != 0 {
+		interestedResources := sets.NewString()
 		for ii := range policy.ExtenderConfigs {
 			glog.V(2).Infof("Creating extender with config %+v", policy.ExtenderConfigs[ii])
 			extender, err := core.NewHTTPExtender(&policy.ExtenderConfigs[ii])
 			if err != nil {
 				return nil, err
 			}
 			extenders = append(extenders, extender)
+			interestedResources.Insert(policy.ExtenderConfigs[ii].InterestedResources...)
 		}
+		predicates.RegisterPodFitsResourcesPredicateMetadataProducer(interestedResources)
```

Comment:
This feels odd. Why not extend the metadataproducer interface?

Reply:
Do you mean ?

```
-type PredicateMetadataProducer func(pod *v1.Pod, nodeNameToInfo map[string]*schedulercache.NodeInfo) PredicateMetadata
+type PredicateMetadataProducer func(pod *v1.Pod, nodeNameToInfo map[string]*schedulercache.NodeInfo, scalarResourcesHandleByExtenders sets.String) PredicateMetadata
```